pkg/secrets/pushtofile/secrets_groups.go

```diff
@@ -0,0 +1,220 @@
+
+			if fileTemplate != "" {
+				fileFormat = FILE_FORMAT_NONE
+			}
+
+			groupInfo := SecretsGroupInfo{
+				Secrets:      secrets,
+				FilePath:     filePath,
+				FileFormat:   fileFormat,
+				FilePerms:    DEFAULT_FILE_PERMS,
+				FileTemplate: fileTemplate,
+			}
+
+			err = validateGroupInfo(groupName, groupInfo)
+			if err != nil {
+				return nil, err
+			}
+
+			secretsGroups[groupName] = groupInfo
+		}
+	}
+
+	return secretsGroups, nil
+}
+
+func parseConjurSecretsPaths(groupName string, secretsPaths string, secretsPathsPrefix string) (SecretsPaths, error) {
+	secrets := make(SecretsPaths)
+
+	unpacked := []interface{}{}
+	if err := yaml.Unmarshal([]byte(secretsPaths), &unpacked); err != nil {
+		return nil, log.RecordedError(messages.CSPFK050E, err.Error(), groupName)
+	}
+
+	insertSecret := func(name string, secretPath string) error {
+		secretPath = path.Clean(secretPath)
+		if secretPath == "." {
+			return log.RecordedError(messages.CSPFK050E, "Invalid secret path", groupName)
+		}
+		secrets[name] = secretsPathsPrefix + secretPath
+
+		return nil
+	}
+
+	for _, secret := range unpacked {
+		switch val := secret.(type) {
+		case string:
+			{
+				name := path.Base(val)
+				if name == "" {
+					return nil, log.RecordedError(messages.CSPFK050E, "Invalid secret name", groupName)
+				}
+
+				if err := insertSecret(name, val); err != nil {
+					return nil, err
+				}
+			}
+
+		case map[string]interface{}:
+			{
+				for alias, secretPath := range val {
+					secretPath, _ := secretPath.(string)
+					if err := insertSecret(alias, secretPath); err != nil {
+						return nil, err
+					}
+				}
+			}
+
+		default:
+			return nil, log.RecordedError(messages.CSPFK050E, "Unknown secret format", groupName)
+		}
+	}
+
+	if len(secrets) == 0 {
+		return nil, log.RecordedError(messages.CSPFK050E, "No valid secrets found", groupName)
+	}
+
+	return secrets, nil
+}
+
+func parseConjurSecretsPathPrefix(groupName string, secretsPathPrefix string) (string, error) {
+	// By default returns the root policy path '/'
+	secretsPathPrefix = path.Clean("/" + secretsPathPrefix)
+	if secretsPathPrefix == "." || secretsPathPrefix == "/" {
+		return "/", nil
+	}
+
+	// Ensure policy path ends with '/'
+	return secretsPathPrefix + "/", nil
+}
+
+func parseFilePath(groupName string, filePath string) (string, error) {
+	// File path must be relative and can not contain a leading '/'
+	if strings.HasPrefix(filePath, "/") {
+		return "", log.RecordedError(messages.CSPFK051E, fmt.Sprintf("%s%s", SECRET_FILE_PATH_PREFIX, groupName))
+	}
+
+	// File path can be a directory (ending in /) or a file name (not ending in /)
+	if strings.HasSuffix(filePath, "/") {
+		filePath = path.Clean(filePath) + "/"
+	} else {
+		filePath = path.Clean(filePath)
+	}
+
+	if filePath == "." {
+		return DEFAULT_FILE_PATH, nil
+	}
+
+	return DEFAULT_FILE_PATH + filePath, nil
+}
+
+func parseFileFormat(groupName string, fileFormat string) (SecretsFileFormat, error) {
```

Comment:
1) Why pass around strings? That's the point of enums
2) IMO better to have a function which takes in the enum value as input and returns the file extension as output instead of attaching it to a struct
3) Changed anyways because who cares